### DISCUSSION TITLE
Aparecer IsADirectoryError en Katas del Modulo 10

### DISCUSSION THREAD

**Discussion by GerardoCabreraH** [LaunchX-InnovaccionVirtual/CursoIntroPython#214] [Q&A] [answered]
He intentado replicar este error de codigo.
```
$ python config.py
Traceback (most recent call last):
  File "/tmp/config.py", line 9, in <module>
    main()
  File "/tmp/config.py", line 3, in main
    configuration = open('config.txt')
IsADirectoryError: [Errno 21] Is a directory: 'config.txt'
```

Pero todavia me sale el mensaje no esta abierto.
```
Couldn't find the config.txt file!
```

Este es mi codigo y me he quedado estancado y no puedo avanzar al siguiente paso del ejercicio de la Kata. 
![paso6](https://user-images.githubusercontent.com/99288879/153735550-3fc3d3a3-7472-4dae-8e8c-2a3daa643d14.png)

y si tengo el archivo, pero si me gustaria que me respondieran como poner exactamente la ruta del archivo para que me de el error IsADirectoryError, por favor.
![Archivo](https://user-images.githubusercontent.com/99288879/153735478-af2139ab-1dc9-462c-8838-58b9f5372f10.png)

**Comment by doguedogue** [comment]
Prueba con este parámetro

f = open("demofile.txt", "r")

**Comment by Stmark1** [comment]
Hola, el error "IsADirectoryError" que tratas de recrear consiste en que tratas de abrir una carpeta como si fuera un archivo.

Lo que ocupas es crear una _carpeta_ ``config.txt`` no un archivo..

**Reply by GerardoCabreraH** [reply]
Hice la carpeta.
![Archivo3](https://user-images.githubusercontent.com/99288879/153735948-2742918e-dddc-4837-9122-3c79b4765426.png)

Pero todavia no le lee el archivo.
![Archivo4](https://user-images.githubusercontent.com/99288879/153736018-212bc290-d5c0-4c37-ac60-2a9d7df4228e.png)

Lo que tuve que hacer es poner en ruta relativa el archivo, pero me sale el error de permisos.
![Archivo7](https://user-images.githubusercontent.com/99288879/153736066-ffa17385-00c6-412f-b284-1642199fef6c.png)

**Reply by doguedogue** [reply]
Ponlo en una ruta por ejemplo

open('c:\\tmp\\config.txt')

**Reply by GerardoCabreraH** [reply]
Ya defini el archivo pero me sale error OS
![Archivo8](https://user-images.githubusercontent.com/99288879/153737137-a7d57ce2-e8d9-4fc7-9482-6b9ed7a31dfc.png)

**Comment by martinGM05** [comment]
Con `IsADirectoryError` no logré que entrara a esa excepción pero si con  `PermissionError`
![Captura de pantalla 2022-02-13 011224](https://user-images.githubusercontent.com/63875704/153743022-43d527b2-63b1-4400-909f-6f8100c61009.png)

**Reply by ShinySick** [reply]
Tampoco e podido replicar el error IsADirectory, unicamente a permissionError

**Comment by doguedogue** [comment, marked as answer]
Esa excepción de IsADirectoryError solo me ha aparecido en linux, wsl y termux.

Me imagino que en windows pasa lo del directorio cuando no tiene una extensión el nombre del directorio que se asemeja a un archivo, ahi solo me ha aparecido el error PermissionError.

![Screenshot_2022-02-14-16-05-39-066_com.termux.jpg](https://user-images.githubusercontent.com/23409026/153955095-f321be78-73a3-4f92-a7b5-fc189dd8f85f.jpg)

**Reply by MauricioNovelo** [reply]
Increible solo por no estar en linux no daba el error correcto, muchas gracias, me fue muy util

**Comment by MartinRodriguezZ** [comment]
Hola, buen dia
intento realizar el mismo error pero no me funciona, me sigue saliendo el mensaje del except primero, intente creando la carpeta config.txt en otra ruta y mandandolo a llamar pero me sale otro error
![image](https://user-images.githubusercontent.com/59885255/154621689-0f97b54a-2623-4968-a640-51f36c4be04a.png)

**Reply by doguedogue** [reply]
para usar la ruta absoluta usa \\ doble diagonal en el path ej.  c:\\Users\\MARZ\\Documents\\config.txt

**Reply by MartinRodriguezZ** [reply]
Muchas gracias si funcionó!
